### PR TITLE
Fix Ankni & Roselex protocol

### PR DESCRIPTION
**Pull request by voidfield101**
Both Ankni and Roselex use the same protocol.

This PR fixes the Ankni protocol as it requires a CRC16 checksum for the handshake.
With those fixes implemented Roselex devices should also now be supported.

### CODE REVIEW

File: buttplug/src/server/device/hardware/communication/btleplug/btleplug_hardware.rs [hunk 1]
```diff
@@ -454,6 +454,6 @@ impl<T: Peripheral> Drop for BtlePlugHardware<T> {
       if let Err(e) = disconnect_fut.await {
         error!("Error disconnecting btleplug device: {:?}", e);
       }
-    })
+    });
```
Comment (blackspherefollower) [inline buttplug/src/server/device/hardware/communication/btleplug/btleplug_hardware.rs:457]: I think this is a no-op change that can be dropped

File: buttplug/buttplug-device-config/buttplug-device-config.json [hunk 2]
```diff
@@ -3869,6 +3869,10 @@
             "tx": "0000fe01-0000-1000-8000-00805f9b34fb",
             "rx": "0000fe02-0000-1000-8000-00805f9b34fb"
           },
+          "0000fffe-0000-1000-8000-00805f9b34fb": {
```
Comment (blackspherefollower) [inline buttplug/buttplug-device-config/buttplug-device-config.json:3872]: The changes to the yaml need to be included too: the JSON is generated from the yaml via `npm run build`

**Review by blackspherefollower — CHANGES_REQUESTED** [review]

File: buttplug/buttplug-device-config/buttplug-device-config.json (hunk 2 above)
Comment (voidfield101) [reply buttplug/buttplug-device-config/buttplug-device-config.json:3872]: Fixed in e9a8f3cc01bad49bfc8523f5d84091b383043aaf.
Completely missed the yaml file.

**Comment by blackspherefollower** [comment]
I've now tested 7 devices that use this protocol and they're all working

**Pull request merged**